src/wallet/wallet.cpp

```diff
@@ -853,6 +853,67 @@ bool CWallet::LoadUnifiedAddressMetadata(const ZcashdUnifiedAddressMetadata &add
                 addrmeta.GetReceiverTypes());
 }
 
+PaymentAddress CWallet::GetPaymentAddressForRecipient(const uint256& txid, const libzcash::RecipientAddress recipient) const {
+    auto recipientsPtr = sendRecipients.find(txid);
+
+    // TODO: We should scrub change addresses from the results here.
+    auto defaultAddress = [&]() -> PaymentAddress {
```

Comment:
This doesn't currently implement the policy we decided on and documented in #5575. At a minimum we should implement the non-contextual baseline policy:
> If the receiver matches a UFVK in the wallet, and we have no information about when this receiver may have been first used, show the UA corresponding to the most recent receiver types policy that includes the receiver's type.

Then add a TODO referencing the issue for the remaining contextual work.

Reply:
#5706